data/fin.lean

```diff
@@ -2,8 +2,21 @@ import data.nat.basic
 
 open fin nat
 
-/-- Embedding of `fin n` in `fin (n+1)` -/
-def raise_fin {n : ℕ} (k : fin n) : fin (n + 1) := ⟨val k, lt_succ_of_lt (is_lt k)⟩
+namespace fin
+
+variable {n : ℕ}
+
+def raise (k : fin n) : fin (n + 1) := ⟨val k, lt_succ_of_lt (is_lt k)⟩
+
+def lower (k : fin (n + 1)) (h : k.val < n) : fin n := ⟨k.val, h⟩
```

Comment:
I don't think we want this function. As it is it's basically `mk`; I guess it would be better if it did something else with the extra element like send n to n-1 and i < n to i. But then we will probably want more control on which element to double, and then it is a different function altogether. I think the easiest thing is just to say there is no `lower` function.